script.js

```diff
@@ -1,4 +1,696 @@
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var shuffledDeck = shuffleCards(deck);
+
+var dealCards = function (shuffledDeck) {
+  if (mode == "compdeal") {
+    var computerCard1 = shuffledDeck.pop();
+    var computerCard2 = shuffledDeck.pop();
+    compHand.push(computerCard1);
+    compHand.push(computerCard2);
+    if (computerCard1.rank == 1 || computerCard2.rank == 1) {
+      mode = "compBlackjackCheck";
+      var compBlackjackText = blackjackCheckFunc(
+        allPlayers,
+        computerCard1,
+        computerCard2,
+        compHand
+      );
+    }
+    mode = "dealmode";
+  }
+  if (dealIndex < noOfPlayers) {
+    var playerCard1 = shuffledDeck.pop();
+    var playerCard2 = shuffledDeck.pop();
+    allPlayers[dealIndex].player.playerHand.push(playerCard1);
+    allPlayers[dealIndex].player.playerHand.push(playerCard2);
+    // if any of the two cards are aces FOR PLAYER
+    playerBlackjack = false;
+    mode = "playerBlackjackCheck";
+    var playerBlackjackText = blackjackCheckFunc(
+      allPlayers,
+      playerCard1,
+      playerCard2,
+      compHand
+    );
+  }
+  dealIndex += 1;
+  if (dealIndex < noOfPlayers) {
+    mode = "dealmode";
+    return `${playerBlackjackText} <br><br> Player ${allPlayers[dealIndex].player.username}, it is your turn.`;
+  } else {
+    mode = "playerDecisionTurn";
+    if (compBlackjack == true) {
+      mode = "compturn";
+      return `${playerBlackjackText}. Since the computer has a blackjack, all players cards will be revealed automatically.`;
+    } else {
+      return `${playerBlackjackText}. Now, Players will choose to hit/stand in order.`;
+    }
+  }
+};
+
+var blackjackCheckFunc = function (allPlayers, card1, card2, compHand) {
+  // if any of the two cards are picture cards or 10 on top of having an ace in hand, it is a blackjack
+  computerCard1 = compHand[0];
+  computerCard2 = compHand[1];
+  var compHandText = `The computer drew:<br>${computerCard1.name} of ${computerCard1.emojiSuit} <br> ${computerCard2.name} of ${computerCard2.emojiSuit}`;
+  // reset player blackjack for future games
+  allPlayers[dealIndex].player.blackjackCheck = false;
+  if (mode == "playerBlackjackCheck") {
+    playerHandText = `You drew: <br>${allPlayers[dealIndex].player.playerHand[0].name} of ${allPlayers[dealIndex].player.playerHand[0].emojiSuit} <br> ${allPlayers[dealIndex].player.playerHand[1].name} of ${allPlayers[dealIndex].player.playerHand[1].emojiSuit}`;
+  }
+  if (mode == "compBlackjackCheck") {
+    computerCard1 = card1;
+    computerCard2 = card2;
+  }
+  if (card1.rank == 1 || card2.rank == 1) {
+    if (
+      card1.rank == 10 ||
+      card1.rank == 11 ||
+      card1.rank == 12 ||
+      card1.rank == 13 ||
+      card2.rank == 10 ||
+      card2.rank == 11 ||
+      card2.rank == 12 ||
+      card2.rank == 13
```

Comment:
would it be easier to change this condition to if card1.rank>=10 || card2.rank>=10 instead of listing out each scenario where the condition would be met?